src/server/_common.py

```diff
@@ -18,12 +20,37 @@ def _get_db() -> Connection:
         g.db = conn
     return g.db
 
+def log_and_raise_exception(message):
+    app.logger.error(message)
+    raise ValidationFailedException(message)
 
 """
 access to the SQL Alchemy connection for this request
 """
 db: Connection = cast(Connection, LocalProxy(_get_db))
 
+@event.listens_for(engine, "before_cursor_execute")
+def before_cursor_execute(conn, cursor, statement, parameters, context, executemany):
+    t = time.time()
+    # TODO: probbly just use one of these two:
+    context._query_start_time = t
+    conn.info.setdefault('query_start_time', []).append(t)
+
+
+@event.listens_for(engine, "after_cursor_execute")
+def after_cursor_execute(conn, cursor, statement, parameters, context, executemany):
+    # this timing info may be suspect, at least in terms of dmbs cpu time...
+    # it is likely that it includes that time as well as any overhead that
+    # come from throttling or flow control on the streamed data, as well as
+    # any row transform/processing time
```

Comment:
typos
```suggestion
    # this timing info may be suspect, at least in terms of dbms cpu time...
    # it is likely that it includes that time as well as any overhead that
    # comes from throttling or flow control on the streamed data, as well as
    # any row transform/processing time
```

Reply:
whoops!  that is definitely referring to the database management system cpu time, and not the dave matthews band's cpu time!

Reply:
true, that band looks organic and not robotic! 🤖